include/OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @project        Open Space Toolkit ▸ Astrodynamics
+/// @file           OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp
+/// @author         Antoine Paletta <antoine.paletta@loftorbital.com>
+/// @license        Apache License 2.0
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_NumericalSolver__
+#define __OpenSpaceToolkit_Astrodynamics_NumericalSolver__
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include <OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.hpp>
+
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+
+#include <boost/numeric/odeint.hpp>
+#include <boost/bind.hpp>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ostk
+{
+namespace astro
+{
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+using ostk::core::types::Real ;
+using ostk::core::types::String ;
+using ostk::core::types::Integer ;
+
+using ostk::physics::time::Instant ;
+using ostk::physics::time::Duration ;
+
+using ostk::astro::flight::system::SatelliteDynamics ;
+
+using namespace boost::numeric::odeint ; // TODO move into function where it is actually used instead of putting it at top
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      Defines a numerical ODE solver that use the Boost Odeint libraries
```

Comment:
Add comment to express the fact that this class will be moved to mathematics later in the future